tarofreighter/wallet.go

```diff
@@ -313,6 +318,57 @@ func (f *AssetWallet) SignVirtualPacket(vPkt *taropsbt.VPacket) ([]uint32,
 	return signedInputs, nil
 }
 
+// PassiveAssetVPackets returns a list of VPacket's for each stale asset that
+// needs to be re-signed and re-anchored.
+func (f *AssetWallet) PassiveAssetVPackets(spendInputs []*taropsbt.VInput,
+	inputCommitment *commitment.TaroCommitment) []*taropsbt.VPacket {
+
+	// Gather non-input assets to re-signing and re-anchor.
+	passiveAssets := inputCommitment.CommittedAssets()
+
+	// Remove input assets (the assets being spent) from list of
+	// assets to re-sign.
+	for i, a := range passiveAssets {
+		for _, inputs := range spendInputs {
+			if a.ID() == inputs.Asset().ID() {
+				passiveAssets = append(
+					passiveAssets[:i],
+					passiveAssets[i+1:]...,
+				)
+				break
+			}
+		}
+	}
+
+	// Create a VPacket for each asset.
+	vPackets := make([]*taropsbt.VPacket, len(passiveAssets))
+	for idx, a := range passiveAssets {
+		vPackets[idx] = &taropsbt.VPacket{
+			Inputs: []*taropsbt.VInput{{
+				PrevID: asset.PrevID{
+					ID: a.ID(),
+				},
+			}},
+			Outputs: []*taropsbt.VOutput{{
+				Amount:   a.Amount,
+				IsChange: true,
+
+				// In this case, the receiver of the output is
+				// also the sender. We therefore set interactive
+				// to true to indicate that the receiver is
+				// aware of the transfer.
+				Interactive: true,
+
+				AnchorOutputIndex: 0,
+				ScriptKey:         a.ScriptKey,
+				Asset:             passiveAssets[idx].Copy(),
+			}},
+		}
```

Comment:
I think after this I still need to do:
```
vPackets[idx].SetInputAsset(0, a, nil)
```
right?

But instead of `nil` I suppose I need to derive and pass through the inclusion proof?

Reply:
No need for the proof in the end. Which makes sense because this vpacket isn't sent to anyone. I figured the proof might have been necessary for validation. But that wasn't the case either. The VM was complaining about something else.